File: FoodOn_backend/src/main/java/com/foodon/foodon/meal/dto/MealInfoResponse.java
package com.foodon.foodon.meal.dto;

import com.foodon.foodon.common.util.BigDecimalUtil;

import java.math.BigDecimal;
import java.util.List;

import static com.foodon.foodon.common.util.BigDecimalUtil.round;

public record MealInfoResponse(
        String imageFileName,
        BigDecimal totalKcal,
        BigDecimal totalCarbs,
        BigDecimal totalProtein,
        BigDecimal totalFat,
        List<MealItemInfo> mealItems
) {
    public static MealInfoResponse from(
            String imageFileName,
            BigDecimal totalKcal,
            BigDecimal totalCarbs,
            BigDecimal totalProtein,
            BigDecimal totalFat,
            List<MealItemInfo> mealItems
    ) {

        return new MealInfoResponse(
                imageFileName,
                round(totalKcal, 0),
                round(totalCarbs, 1),
                round(totalProtein, 1),
                round(totalFat, 1),
                mealItems
        );
    }
}

File: android-app/app/src/main/java/com/swallaby/foodon/data/food/remote/dto/response/MealInfoResponse.kt
package com.swallaby.foodon.data.food.remote.dto.response

import com.swallaby.foodon.domain.food.model.MealInfo
import com.swallaby.foodon.domain.food.model.MealItem

data class MealInfoResponse(
     val imageFileName: String = "",
     val totalCarbs: Double = 0.0,
     val totalFat: Double = 0.0,
     val totalKcal: Double = 0.0,
     val totalProtein: Double = 0.0,
     val mealItems: List<MealItem> = emptyList(),
)

fun MealInfoResponse.toDomain(): MealInfo = MealInfo(
    imageFileName = imageFileName,
    totalCarbs = totalCarbs,
    totalFat = totalFat,
    totalKcal = totalKcal.toInt(),
    totalProtein = totalProtein,
    mealItems = mealItems
)